dash/include/dash/halo/Halo.h

```diff
@@ -11,55 +11,150 @@
 
 namespace dash {
 
-template <dim_t NumDimensions>
-class Stencil : public Dimensional<int16_t, NumDimensions> {
+/**
+ * Stencil point with raletive coordinates for N dimensions
+ * e.g. StencilPoint<2>(-1,-1) -> north west
+ */
+template <dim_t NumDimensions, typename CoeffT = double>
+class StencilPoint : public Dimensional<int16_t, NumDimensions> {
 private:
   using Base_t = Dimensional<int16_t, NumDimensions>;
 
 public:
   // TODO constexpr
-  Stencil() {
+  /**
+   * Default Contructor
+   *
+   * All stencil point values are 0 and default coefficient = 1.0.
+   */
+  StencilPoint() {
     for(dim_t i(0); i < NumDimensions; ++i) {
       this->_values[i] = 0;
     }
   }
 
+  /**
+   * Constructor
+   *
+   * Custom stencil point values for all dimensions and default
+   * coefficient = 1.0.
+   */
+  template <typename... Values>
+  constexpr StencilPoint(
+    typename std::enable_if<sizeof...(Values) == NumDimensions - 1,
+                            int16_t>::type value,
+    Values... values)
+  : Base_t::Dimensional(value, (int16_t) values...) {}
+
+  /**
+   * Constructor
+   *
+   * Custom values and custom coefficient.
+   */
   template <typename... Values>
-  constexpr Stencil(int16_t value, Values... values)
-  : Base_t::Dimensional(value, values...) {}
+  constexpr StencilPoint(
+    typename std::enable_if<sizeof...(Values) == NumDimensions - 1,
+                            CoeffT>::type coefficient,
+    int16_t                               value, Values... values)
+  : Base_t::Dimensional(value, (int16_t) values...), _coefficient(coefficient) {
+  }
 
   // TODO as constexpr
+  /**
+   * Returns maximum distance to center over all dimensions
+   */
   int max() const {
     int16_t max = 0;
     for(dim_t i(0); i < NumDimensions; ++i)
       max = std::max((int) max, (int) std::abs(this->_values[i]));
     return max;
   }
-};  // Stencil
 
-template <dim_t NumDimensions, std::size_t NumStencilPoints>
+  /**
+   * Returns the coefficient for this stencil point
+   */
+  CoeffT coefficient() const { return _coefficient; }
+
+private:
+  CoeffT _coefficient = 1.0;
+};  // StencilPoint
+
+/**
+ * A collection of stencil points (\ref Stencil)
+ * e.g. StencilSpec<dash::StencilPoint<2>, 2,2>({StencilPoint<2>(-1,0),
+ * StencilPoint<2>(1,0)}) -> north and south
+ */
+template <typename StencilPointT, dim_t NumDimensions,
+          std::size_t NumStencilPoints>
```

Comment:
Is there a valid case for `StencilPointT<N>` and `N != NumDimensions`? If not, can't we default `StencilPointT` to `StencilPoint<NumDimensions>` to save the user some typing?

Reply:
correct